src/lib/netlist/plib/ppmf.h

```diff
@@ -28,6 +28,91 @@
 ///
 ///
 
+/// \brief Enable experimental code on Visual Studio builds and VS clang llvm builds
+///
+/// This enambles experimal code which uses optimized builds the
+/// PPMF_TYPE_INTERNAL_MSC path also for complex (struct/union) return types.
+/// This currently depends on whether the code can adequately determine on
+/// x64 builds if the return type is returned through registers or passed as a
+/// second argument as a pointer to the member function.
+///
+/// ppmf uses a temporary storage for the return value. This causes for large sized
+/// return types a copy overhead. It would be easier if we would be able to obtain
+/// the RDX register on entry to call. On MSVC this seems not to be possible since on
+/// x64 inline assembly is not supported. An alternative would be to take the
+/// address of the first argument and substract sizeof(void *) from it. This would
+/// give us the address of the pointer to the return buffer. For member functions
+/// which do not have at least one parameter the approach does not work.
+/// In addition, this code has to be compiled with /w34716 to not abort on a
+/// missing return in the call function. The approach can be checked in godbolt.com
+/// with the following code:
```

Comment:
This won’t work properly:
* When optimisation is on, there’s no guarantee the compiler will spill the pointer to the return area into the home space on the stack before you try taking the address of the first formal parameter.  If the value is left in RDX, you won’t be able to obtain from memory.
* Taking the address of the first formal parameter will force it to be spilled if it was eligible to be passed in a register, but if it was ineligible to be passed in a register this will merely obtain the value from R8 which points to the area where the caller placed the value.
* Although the value passed in R8 is eligible to be spilled to the home space on the stack, there’s no guarantee this is where it will actually be spilled to – when optimisation is enabled, it could theoretically be spilled anywhere in the function’s stack frame.
* If the first formal parameter is a reference, taking its address just obtains the value from R8, without forcing it to be spilled.  It doesn’t obtain the address that it’s spilled to.
* If the first formal parameter is a floating-point or vector type, it is ineligible to be spilled to the R8 home space.

There are simply too many ways for this to go wrong.

Reply:
I have updated the comment and removed the code example. I now have committed my changes:

- Moved constructor and member functions outside the class declaration
  with the exception of "call" member functions.
- Updated comment on accessing RDX
- Added more test cases

Thanks for pointing out the outside of class changes. I am sure there are cases in the code which I will take care of.